app/services/api/webflow-helpers.ts

```diff
@@ -178,27 +185,40 @@ const convertScheduleToCardProps = (schedule: ScheduledEvent): ScheduleCardProps
         eventTitle: "workshop",
         heading: workshop?.name,
         subheading: workshop?.["instructor-info"]?.name,
-        sources: [workshop?.["instructor-info"]?.["speaker-photo"].url],
+        sources: [workshop?.["instructor-info"]?.["speaker-photo"]?.url],
         level: workshop?.level,
         id: schedule._id,
       }
+    default:
+      throw new Error(`Unknown schedule type '${schedule.type}' for event '${schedule.name}'`)
   }
 }
 
 // [NOTE] util function that might be needed in the future
 export const convertScheduleToScheduleCard = (
   scheduleData: ScheduledEvent[],
-  day: string,
+  day: Day,
 ): ScheduleCardProps[] => {
-  const daySchedule: ScheduledEvent[] = groupBy("day")(scheduleData ?? [])?.[day] ?? []
-  return daySchedule.sort(sortByTime).map(convertScheduleToCardProps)
+  // 1. Get the schedule for the current day
+  const groupByDay = groupBy("day")
+  const schedules = groupByDay(scheduleData)
+  const daySchedule: ScheduledEvent[] = schedules[day] ?? []
+
+  // 2. Sort the schedule by time
+  const sortedSchedule: ScheduledEvent[] = daySchedule.sort((a, b) =>
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    sortByTime(a["day-time"]!, b["day-time"]!),
```

Comment:
This is due to a weird issue with Zod where array properties are set to optional when `strict: true` is not set in `tsconfig.json`

Reply:
The other changes are attempts during debugging